java-checks/src/main/java/org/sonar/java/filters/LombokFilter.java

```diff
@@ -181,12 +182,34 @@ public void visitAssignmentExpression(AssignmentExpressionTree tree) {
   }
 
   private static boolean usesAnnotation(ClassTree classTree, List<String> annotations) {
+    return usesAnnotation(classTree, annotations, !CHECK_LOCAL_NAME);
+  }
+
+  private static final boolean CHECK_LOCAL_NAME = true;
```

Comment:
If the value is always `true` then we can remove this field that introduces dependencies between static methods and fields.
```suggestion
```